### ISSUE TITLE
[FEATURE_REQUEST] Support ControlNet API for automatic1111

### ISSUE THREAD

**Issue by slayermaster** [SillyTavern/SillyTavern#1305]
### Have you searched for similar requests?

Yes

### Is your feature request related to a problem? If so, please describe.

I've always disliked how the generated pictures have different character look compared to the one I'm chatting with.

### Describe the solution you'd like

Support ControlNet API and use the "Reference" preprocessor, using the profile picture of the current character or user can use custom image and it only applies based on character.

### Describe alternatives you've considered

Using LoRA's for different characters, but sometimes there is no character LoRAs available.

### Priority

Low (Nice-to-have)

### Is this something you would be keen to implement?

None

**Comment by rosx27** [comment]
yes! this! also, can we include roop/reactor/ipadapter? so that it will take the photo of the character and use that as reference in image generation. the prompt would be much more manageable since we only have to describe the scene or actions of the character.

**Comment by slayermaster** [comment]
[vincedundee/ST-WebUI-Diffusion](https://github.com/vincedundee/ST-WebUI-Diffusion)
found this here @rosx27

**Comment by kainatquaderee** [comment]
> [vincedundee/ST-WebUI-Diffusion](https://github.com/vincedundee/ST-WebUI-Diffusion) found this here @rosx27

i  think user got deleted or something cant ACCess it. can you provide your file of this repo or something I want to rebuild it.

**Comment by slayermaster** [comment]
> i think user got deleted or something cant ACCess it. can you provide your file of this repo or something I want to rebuild it.

I made a mirror here. [slayermaster/ST-WebUI-Diffusion](https://github.com/slayermaster/ST-WebUI-Diffusion)
